### PR TITLE
ddl: check foreign key constraint when drop,modify,change column

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
* Check foreign key constraint when drop,modify,change column.

parser PR: https://github.com/pingcap/parser/pull/675

Otherwise, select `information_schema.KEY_COLUMN_USAGE` will panic.

```sql
create table t1 (a int,b int);
create table t2 (c int,d int,foreign key (d) references t1 (b));
alter table t2 drop column d;
select count(*) from information_schema.KEY_COLUMN_USAGE;
```

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch

Release note

 - Fix issue of select KEY_COLUMN_USAGE table panic cause by the foreign key.

### CODE REVIEW

File: ddl/column.go [hunk 1]
```diff
@@ -535,6 +535,17 @@ func isColumnWithIndex(colName string, indices []*model.IndexInfo) bool {
 	return false
 }
 
+func isColumnWithForeignKey(colName string, fkInfos []*model.FKInfo) *model.FKInfo {
+	for _, fkInfo := range fkInfos {
```
Comment (AilinKid) [inline ddl/column.go]: Feel like we can change the name to `isForeignKeyWithColumn` or `isColumnInForeignKey`

File: ddl/db_test.go [hunk 2]
```diff
@@ -2306,7 +2306,18 @@ func (s *testDBSuite2) TestTableForeignKey(c *C) {
...
+	s.tk.MustExec("create table t4 (c int,d int,foreign key (d) references t1 (b));")
+	failSQL = "alter table t4 drop column d"
+	s.tk.MustGetErrCode(failSQL, mysql.ErrFkColumnCannotDrop)
+	// Test change column with foreign key.
+	failSQL = "alter table t4 change column d e bigint;"
+	s.tk.MustGetErrCode(failSQL, mysql.ErrReferencedForeignKey)
+	// Test modify column with foreign key.
+	failSQL = "alter table t4 modify column d bigint;"
+	s.tk.MustGetErrCode(failSQL, mysql.ErrReferencedForeignKey)
+	s.tk.MustQuery("select count(*) from information_schema.KEY_COLUMN_USAGE;")
```
Comment (AilinKid) [inline ddl/db_test.go]: this case for what?

File: ddl/column.go (hunk 1 above)
Comment (djshow832) [reply ddl/column.go]: Return value is `FKInfo` but not bool, so the function name should begin with `get`.

File: ddl/db_test.go (hunk 2 above)
Comment (crazycs520) [reply ddl/db_test.go]: See description, before this pr, select `KEY_COLUMN_USAGE ` will panic.

Comment (AilinKid) [reply ddl/db_test.go]: got it.

**Comment by crazycs520** [comment]
@reafans, Already add test for the case you said.

**Comment by reafans** [comment]
LGTM

**Pull request merged**